packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/template/ProxyTemplate.java

```diff
@@ -132,16 +132,12 @@ public final void write(Proxy restAPI, JavaClass classBlock) {
                         }
                     }
 
-                    if (!settings.isDataPlaneClient() || !settings.isAzureV1() || isExceptionCustomized()) {
-                        // write @UnexpectedResponseExceptionType
-
-                        if (restAPIMethod.getUnexpectedResponseExceptionTypes() != null) {
-                            writeUnexpectedExceptions(restAPIMethod, interfaceBlock);
-                        }
-
-                        if (restAPIMethod.getUnexpectedResponseExceptionType() != null) {
-                            writeSingleUnexpectedException(restAPIMethod, interfaceBlock);
-                        }
+                    // write @UnexpectedResponseExceptionType
+                    if (restAPIMethod.getUnexpectedResponseExceptionTypes() != null) {
+                        writeUnexpectedExceptions(restAPIMethod, interfaceBlock);
+                    }
+                    if (restAPIMethod.getUnexpectedResponseExceptionType() != null) {
+                        writeSingleUnexpectedException(restAPIMethod, interfaceBlock);
                     }
```

Comment:
Change here has a risk of affect other flavors.

For DPG, it actually gets in the condition, as `isExceptionCustomized() == true`
azureV2 should be true as well.
Hence there seems not much about condition on DPG.

I would use the sync PR on autorest.java to double check whether it would affect any existing tests.